src/Core/Constants.cs

```diff
@@ -23,6 +23,40 @@ public static class Constants
     public const string CipherKeyEncryptionMinimumVersion = "2023.9.1";
 }
 
+public static class AuthConstants
+{
+    public static readonly RangeConstant PBKDF2_ITERATIONS = new(100_000, 2_000_000, 600_000);
+
+    public static readonly RangeConstant ARGON2_ITERATIONS = new(1, 10, 3);
+    public static readonly RangeConstant ARGON2_MEMORY = new(15, 1024, 64);
+    public static readonly RangeConstant ARGON2_PARALLELISM = new(1, 16, 4);
+
+}
+
+public class RangeConstant
+{
+    public int Default { get; }
+    public int Min { get; }
+    public int Max { get; }
+
+    public RangeConstant(int min, int max, int def)
+    {
```

Comment:
❓ Should we also verify that the `min` is not greater than the `max`?
```C#
 if (min > max){
    throw new ArgumentException("Minimum value should not be greater than the maximum value.");
}
```

Reply:
Yea probably.